crates/proposer-service/src/lib.rs

```diff
@@ -62,31 +62,24 @@ impl<L1Rpc>
         let network_prover = new_network_prover(config.client.sp1_cluster_endpoint.as_str())
             .map_err(Error::UnableToCreateNetworkProver)?;
 
+        let aggregation_vkey_hash = Self::extract_aggregation_vkey(AGGREGATION_ELF);
+
         Ok(Self {
             l1_rpc,
             client: Arc::new(proposer_client::client::Client::new(
                 proposer_rpc_client,
                 network_prover,
                 Some(config.client.proving_timeout),
             )?),
-            aggregation_vkey_hash: AGGREGATION_VKEY_HASH,
+            aggregation_vkey_hash,
         })
     }
-}
 
-fn check_aggregation_vkey(
-    proof: &AggregationProof,
-    expected_vkey_hash: VKeyHash,
-) -> Result<(), Error> {
-    let vkey_hash = VKeyHash::from_vkey(&proof.vk);
-    if vkey_hash != expected_vkey_hash {
-        return Err(Error::AggregationVKeyMismatch {
-            got: vkey_hash,
-            expected: expected_vkey_hash,
-        });
```

Comment:
Since this error is no longer produced, it can be removed from the corresponding enum.